include/grpc/grpc_security.h

```diff
@@ -320,6 +356,38 @@ GRPCAPI grpc_server_credentials *grpc_ssl_server_credentials_create_ex(
     grpc_ssl_client_certificate_request_type client_certificate_request,
     void *reserved);
 
+typedef struct grpc_ssl_server_credentials_options
+    grpc_ssl_server_credentials_options;
+
+/** Creates an options object using a certificate config. Use this method when
+   the certificates and keys of the SSL server will not changes during the
+   server's lifetime. */
+GRPCAPI grpc_ssl_server_credentials_options *
+grpc_ssl_server_credentials_create_options_using_config(
```

Comment:
One small thing that I find a bit unfortunate here is that it's a bit heavy to use. E.g.:

```c
grpc_ssl_pem_key_cert_pair key_cert_pair = {my_cert_chain, my_priv_key};
grpc_ssl_server_certificate_config *config = grpc_server_certificate_config_create(NULL, &key_cert_pair, 1);
grpc_ssl_server_credentials_options *options = grpc_server_certificate_create_options_with_config(config);
grpc_server_crededentials *creds = grpc_ssl_server_credentials_create_with_options(options);
grpc_ssl_server_credentials_options_release(options);
grpc_ssl_server_certificate_config(config);
```
Maybe we could skip the creation of the config in this case and have the `options_with_config` directly take the keys as an argument as opposed to the config. Just a thought... I'll leave that up to you. 
The proposed change has the drawback that we are not passing a real "config" now.  Yet another solution is to have the options take ownership of the config (which would make sense to me).

Reply:
I think we can add a wrapper later to make it slightly more convenient to use these.... with the move to c++ i think things like releasing/freeing can be done by scoped/smart pointers etc... and it wouldn't be that bad at all.

can we do this later? :)

Reply:
Yes, we'll postpone further changes here.

Reply:
Hmm, actually, I will review whether we should take ownership of the options.

Reply:
oh ok.... do you plan to discuss with @jboeuf off-line first like you guys did before (to minimize the round-trips here :) )